tests/generic_config_updater/test_aaa.py

```diff
@@ -0,0 +1,642 @@
+import logging
+import pytest
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.generic_config_updater.gu_utils import apply_patch, expect_op_success, expect_op_failure
+from tests.generic_config_updater.gu_utils import generate_tmpfile, delete_tmpfile
+from tests.generic_config_updater.gu_utils import create_checkpoint, delete_checkpoint, rollback_or_reload
+
+pytestmark = [
+    pytest.mark.topology('t0'),
+]
+
+logger = logging.getLogger(__name__)
+
+AAA_CATEGORY = ["authentication", "authorization", "accounting"]
+
+@pytest.fixture(autouse=True)
+def setup_env(duthosts, rand_one_dut_hostname):
+    """
+    Setup/teardown fixture for each loopback interface test.
+    rollback to check if it goes back to starting config
+
+    Args:
+        duthosts: list of DUTs.
+        rand_selected_dut: The fixture returns a randomly selected DuT.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+
+    create_checkpoint(duthost)
+
+    yield
+
+    try:
+        logger.info("Rolled back to original checkpoint")
+        rollback_or_reload(duthost)
+    finally:
+        delete_checkpoint(duthost)
+
+def get_aaa_sub_options_value(duthost, aaa_type, option):
+    """ Verify if AAA sub type's options match with expected value
+
+    Sample output:
+    admin@vlab-01:~$ show aaa | grep -Po "AAA authentication login \K.*"
+    local (default)
+    """
+    output = duthost.shell('show aaa | grep -Po "AAA {} {} \K.*"'.format(aaa_type, option))
+
+    pytest_assert(not output['rc'],
+        "Failed to grep AAA {}".format(option)
+    )
+    return output['stdout']
+
+def aaa_add_init_config_without_table(duthost):
+    """ Add initial config not containing AAA table
+
+    Configure to default setting which doesn't contain AAA table
+    Sample configDB without table:
+    admin@vlab-01:~$ show run all | grep AAA
+    admin@vlab-01:~$
+    """
+    cmds = 'sonic-db-cli CONFIG_DB keys "AAA|*" | xargs -r sonic-db-cli CONFIG_DB del'
```

Comment:
>cmds [](http://example.com/codeflow?start=4&length=4)

Does it work if the Redis has no `AAA|*` tables?

Reply:
Yeah. It is the shortcut for `--no-run-if-empty`
```
admin@vlab-01:~$ xargs --help
...
-r, --no-run-if-empty        if there are no arguments, then do not run COMMAND;
```